packages/Operators/src/PointCloud/DTK_NodeToNodeOperator_impl.hpp

```diff
@@ -137,12 +143,14 @@ void NodeToNodeOperator<DIM>::setupImpl(
             // If we have a neighbor then there should be only 1.
             DTK_CHECK( 1 == pairings.childCenterIds( i ).size() );
 
-            // Check that our neighbor node has the same coordinates.
-            DTK_CHECK(
+            // Check that our neighbor node has the same coordinates if the
+            // nodes are supposed to match.
+            DTK_REMEMBER( bool distance_check =
                 std::abs( EuclideanDistance<DIM>::distance(
-                    dist_sources( DIM * pairings.childCenterIds( i )[0], DIM )
-                        .getRawPtr(),
-                    target_centers( DIM * i, DIM ).getRawPtr() ) ) < 1.0e-14 );
+                              dist_sources( DIM * pairings.childCenterIds( i )[0], DIM )
+                              .getRawPtr(),
+                              target_centers( DIM * i, DIM ).getRawPtr() ) ) < 1.0e-14 );
+            DTK_CHECK( (distance_check && d_matching_nodes) || !d_matching_nodes );
```

Comment:
I thought about that but didnt do it for two reasons:

1) I think separating it makes it more clear what check is being performed
2) This check is performed in DBC so I am not worried about performance.